node-fs-readfile/cat.js

```diff
@@ -0,0 +1,20 @@
+import { readFile } from 'node:fs/promises';
+
+const arrayOfFiles = process.argv.splice(2);
+const promises = arrayOfFiles.map(async (fileName) => {
+  try {
+    const filePath = new URL(fileName, import.meta.url);
+    const content = await readFile(filePath, { encoding: 'utf8' });
+    return content;
+  } catch (error) {
+    console.log(error);
+  }
+});
+
+await Promise.all(promises)
+  .then((contents) => {
```

Comment:
Oops. You don't want to use `.then` and `.catch` if you are using `await`. Just say
`const contents = await Promise.all(promises);` (wrapped in a try/catch)
then you can use your for-loop to log the contents.